src/db_client/cluster.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{collections::HashMap, sync::Arc};
+
+use async_trait::async_trait;
+use dashmap::DashMap;
+use futures::future::join_all;
+
+use super::{standalone::StandaloneImpl, DbClient};
+use crate::{
+    errors::{should_refresh, ClusterWriteError},
+    model::{
+        request::QueryRequest,
+        route::Endpoint,
+        write::{WriteRequest, WriteResponse},
+        QueryResponse,
+    },
+    router::Router,
+    rpc_client::{RpcClientImpl, RpcClientImplBuilder, RpcContext},
+    Error, Result,
+};
+
+/// Client for ceresdb of cluster mode.
+pub struct ClusterImpl<R: Router> {
+    router: R,
+    standalone_pool: StandalonePool,
+}
+
+#[async_trait]
+impl<R: Router> DbClient for ClusterImpl<R> {
+    async fn query(&self, ctx: &RpcContext, req: &QueryRequest) -> Result<QueryResponse> {
+        if req.metrics.is_empty() {
+            return Err(Error::Unknown(
+                "Metrics in query request can't be empty in cluster mode".to_string(),
+            ));
+        }
+
+        let endpoint = match self.router.route(&req.metrics, ctx).await {
+            Ok(mut eps) => {
+                if let Some(ep) = eps[0].take() {
+                    ep
+                } else {
+                    return Err(Error::Unknown(
+                        "Metric doesn't have corresponding endpoint".to_string(),
+                    ));
+                }
+            }
+            Err(e) => {
+                return Err(e);
+            }
+        };
+
+        let clnt = self.standalone_pool.get_or_create(&endpoint).clone();
```

Comment:
`client` is fine, this kind of abbr is not necessary in my opinion.

Reply:
Done.